src/app/(admin)/admin/_lib/validationSchemas/specialistSchema.js

```diff
@@ -0,0 +1,79 @@
+import { z } from 'zod';
+import { Gender } from '@prisma/client';
+import {
+  MESSAGES,
+  createValidationSchema,
+  specialistCore,
+  zCreateAddressSchema,
+  zEditAddressSchema,
+  zInteger,
+  zString,
+  zStringArray,
+  zStringWithMax,
+} from './specialistCommonSchemas';
+
+// ------------------ COMMON SECTION ---------------------
+
+const zSpecialistSchema = specialistCore.extend({
+  surname: zStringWithMax.nullish(),
+  gender: zString.refine(val => Object.values(Gender).includes(val), {
+    message: MESSAGES.unacceptableValue,
+    invalid_type_error: 'Неприпустиме значення',
+  }),
+  yearsOfExperience: zInteger,
+  description: zString.nullish(),
+});
+
+const restProps = zSpecialistSchema.extend({
+  addresses: zCreateAddressSchema.array().default([]),
+});
+
+// ------------------ CREATE SECTION ---------------------
+
+const createDefaultProps = z.object({
+  lastName: zStringWithMax,
+  firstName: zStringWithMax,
+  specializations: zStringArray,
+});
+
+const activeSpecialistSchema = restProps.extend({
+  isActive: z.literal(true),
+});
+
+const draftSpecialistSchema = restProps.partial().extend({
+  isActive: z.literal(false),
+});
+
+export const specialistSchemaUnion = z.discriminatedUnion('isActive', [activeSpecialistSchema, draftSpecialistSchema]);
+export const specialistCreateValidationSchema = createValidationSchema(specialistSchemaUnion, createDefaultProps);
+
+// ------------------ EDIT SECTION ---------------------
```

Comment:
same here with splitting edit/create

Reply:
Thanks for comment! The reason is explained above.